api/lib/src/lp.rs

```diff
@@ -241,20 +243,39 @@ pub trait LpApi: SignedExtrinsicApi {
 		id: OrderId,
 		option_tick: Option<Tick>,
 		amount_change: IncreaseOrDecrease<AssetAmount>,
+		details: Option<OrderScheduleDetails<BlockNumber>>,
 	) -> Result<Vec<types::LimitOrder>> {
 		// Submit the mint order
-		let (_tx_hash, events, ..) = self
-			.submit_signed_extrinsic(pallet_cf_pools::Call::update_limit_order {
-				sell_asset,
-				buy_asset,
-				id,
-				option_tick,
-				amount_change,
-			})
-			.await
-			.until_in_block()
-			.await?;
-
+		let events = if let Some(details) = details {
+			let (_tx_hash, events, ..) = self
+				.submit_signed_extrinsic(pallet_cf_pools::Call::schedule {
```

Comment:
You would dedup some codeby moving the update_limit_order construction outside the `if let Some()`

Reply:
You could then dedup more code, by introducing a function scheduled_or_immediate (or something like that) that takes a call, and an Option<OrderScheduleDetail>, and either schedules it or does the call immediately.